enhancements/network/on-prem-service-load-balancers.md

```diff
@@ -0,0 +1,405 @@
+outside of the cluster, verifiable using existing e2e tests that make use of
+service load balancers.
+
+#### Story 1 - Easy Use with a Small Cluster
+
+As an administrator of a small cluster that resides entirely on a single layer
+2 domain, I would like to configure one or more ranges of IP addresses from my
+network that the cluster is free to use for Service Load Balancers.  I do not
+want to do any extra configuration in my network infrastructure beyond just
+ensuring that the configured ranges of addresses are not used elsewhere.
+
+MetalLB can do this today.
+
+#### Story 2 - BGP Integration
+
+As an administrator of a larger cluster with Nodes that reside on multiple L2
+network segments, I would like to configure one or more ranges of IP addresses
+from my network that the cluster is free to use for Service Load Balancers.  I
+would like my cluster to peer with my BGP infrastructure to advertise the
+current location of IP addresses allocated to Service Load Balancers.
+
+MetalLB can do this today.
+
+#### Story 3 - Larger Clusters without BGP
+
+As an administrator of a larger cluster with Nodes that reside on multiple L2
+network segments, I would like to configure one or more ranges of IP addresses
+from my network that the cluster is free to use for Service Load Balancers.  I
+do not have BGP infrastructure available or I'm not willing to have my cluster
+peer with my BGP infrastructure.  I would like to configure awareness of which
+subsets of my Nodes can use which pools of IP addresses since not every Node
+has physical connectivity to the same L2 networks.
+
+Note that MetalLB does not offer this today.  There has been some discussion
+about related functionality:
+* https://github.com/metallb/metallb/issues/605
+* https://github.com/metallb/metallb/pull/502
+
+### Implementation Details/Notes/Constraints
+
+#### Upstream Engagement
+
+The first step of implementation is to invest in the upstream project.  We
+should have one or more engineers engage with the project to handle issues,
+review pull requests, and contribute bug fixes or enhancements.  As part of
+this process we should continue our technical due diligence with testing and
+reviewing code to increase our confidence in choosing this solution.
+
+One area we could contribute immediately is with setting up upstream CI.  The
+project does not appear to run any CI today.
+
+[kind](https://github.com/kubernetes-sigs/kind/) is a good basis for CI of
+community kubernetes-ecosystem projects. It is usable with the built-in free
+github CI support rather than needing someone to be paying for test
+infrastructure elsewhere. It would even allow testing both L2 and L3 mode. It
+doesn't matter what protocols the actual underlying network supports if you're
+doing all of your testing in a virtual network built on top of it.
+
+#### Operator
+
+We must also create an operator for MetalLB.  We should develop an operator
```

Comment:
See my comment https://github.com/openshift/enhancements/pull/356/files#r434739926 and the `Packet` controller https://github.com/packethost/packet-ccm